delocate/tools.py

```diff
@@ -148,6 +148,30 @@ def _run(
         ) from exc
 
 
+# Mach-O magic numbers. See:
+# https://github.com/apple-oss-distributions/xnu/blob/5c2921b07a2480ab43ec66f5b9e41cb872bc554f/EXTERNAL_HEADERS/mach-o/loader.h#L65
+# https://github.com/apple-oss-distributions/cctools/blob/658da8c66b4e184458f9c810deca9f6428a773a5/include/mach-o/fat.h#L48
+MACHO_MAGIC = (
+    0xFEEDFACE.to_bytes(4, "little"),  # MH_MAGIC
+    0xFEEDFACE.to_bytes(4, "big"),  # MH_MAGIC
+    0xFEEDFACF.to_bytes(4, "little"),  # MH_MAGIC_64
+    0xFEEDFACF.to_bytes(4, "big"),  # MH_MAGIC_64
+    0xCAFEBABE.to_bytes(4, "big"),  # FAT_MAGIC (always big-endian)
+    0xCAFEBABF.to_bytes(4, "big"),  # FAT_MAGIC_64 (always big-endian)
+)
+
+
+def _is_macho_file(filename: str) -> bool:
```

Comment:
Would you mind running this over some actual `.so` files?   Maybe use the files I added in this branch?  https://github.com/matthew-brett/delocate/tree/test-files 

See the `delocate/tests/data` directory for the `np-*.dylib` files.

Reply:
I added a test in [640d0cb](https://github.com/matthew-brett/delocate/pull/172/commits/640d0cb12859a16a8c79e6c445d0afb37ac36c95) that only the mach-o files in test-data are counted as such. If you merge your test-files branch, they'll be automatically tested. I also ran the test manually with those files.

Of course, `_is_macho_file` just looks at the first 4 bytes so non-macho files may be reported as macho files. (I think java bytecode files also use the 0xcafebabe magic number.) That's fine though, it just means more calls to `otool`.